assets/src/edit-story/app/uploader/useUploader.js

```diff
@@ -124,6 +116,31 @@ function useUploader() {
         ..._additionalData,
       };
 
+      if (canTranscodeFile) {
+        try {
+          // TODO: Only transcode & optimize video if needed (criteria TBD).
+          const newFile = await transcodeVideo(file);
+          return uploadMedia(newFile, additionalData);
```

Comment:
```suggestion
          const  additionalDataVideo = {
               ...additionalData, 
               media_source: 'video-optimization',
          };
          return uploadMedia(newFile, additionalDataVideo);
```

We need to get the media source here.